apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm, type SubmitHandler } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import * as Yup from 'yup';
+import { CheckboxField } from '../../shared/forms/CheckboxField';
+import { Form } from '../../shared/forms/Form';
+import { TextField } from '../../shared/forms/TextField';
+import ExternalLink from '../external-link';
+import { Button } from '_app/shared/ButtonUI';
+import { ToS_LINK } from '_src/shared/constants';
+
+const formSchema = Yup.object({
+	password: Yup.string().required('Required'),
+	confirmedPassword: Yup.string().required('Required'),
+	acceptedTos: Yup.boolean().required().oneOf([true]),
+	enabledAutolock: Yup.boolean(),
+});
+
+type FormValues = Yup.InferType<typeof formSchema>;
+
+type ProtectAccountFormProps = {
+	submitButtonText: string;
+	cancelButtonText: string;
+	onSubmit: SubmitHandler<FormValues>;
+};
+
+export function ProtectAccountForm({
+	submitButtonText,
+	cancelButtonText,
+	onSubmit,
+}: ProtectAccountFormProps) {
+	const form = useForm({
+		mode: 'all',
+		defaultValues: {
+			password: '',
+			confirmedPassword: '',
+			acceptedTos: false,
+			enabledAutolock: true,
+		},
+		resolver: yupResolver(formSchema),
```

Comment:
could be nice to attempt to achieve the equivalent of `useZodForm` with `Yup` (or ditch Yup at some point soon)

Reply:
yeah ditching `Yup` would be nice, I just didn't want to spend the time re-writing all of the validation logic haha